app/src/main/java/org/openhdp/hdt/ui/settings/SettingsViewModel.kt

```diff
@@ -0,0 +1,118 @@
+package org.openhdp.hdt.ui.settings
+
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.openhdp.hdt.data.StopwatchRepository
+import org.openhdp.hdt.data.entities.Stopwatch
+import org.openhdp.hdt.ui.base.BaseViewModel
+import org.openhdp.hdt.ui.settings.export.ExportCategoriesUseCase
+import org.openhdp.hdt.ui.settings.export.ExportStopwatchesUseCase
+import org.openhdp.hdt.ui.settings.export.ExportTimestampsUseCase
+import timber.log.Timber
+
+class SettingsViewModel @ViewModelInject constructor(
+    private val stopwatchRepository: StopwatchRepository,
+    private val exportStopwatchesUseCase: ExportStopwatchesUseCase,
+    private val exportCategoriesUseCase: ExportCategoriesUseCase,
+    private val exportTimestampsUseCase: ExportTimestampsUseCase,
+    private val startOfDayUseCase: StartOfDayUseCase
+) : BaseViewModel<SettingsViewState>() {
+
+    fun initialize() {
+        val startOfDay = startOfDayUseCase.getCurrentStartOfDay()
+        setState(
+            SettingsViewState.Display(startOfDay.hours, startOfDay.minutes)
+        )
+    }
+
+    fun onEvent(event: SettingsEvent) = when (event) {
+        SettingsEvent.ExportStopwatches -> {
+            exportStopwatchesData()
+        }
+        is SettingsEvent.ExportTimestamps -> {
+            exportStopwatchTimestamps(event.stopwatch)
+        }
+        is SettingsEvent.ChangeStartOfDay -> {
+            startOfDayUseCase.saveStartOfDay(event.hourOfDay, event.minute)
+            initialize()
+        }
+        SettingsEvent.RequestGenericExportDialog -> {
+            requestExportDialog()
+        }
+        SettingsEvent.ExportCategories -> {
+            requestExportCategories()
+        }
+    }
+
+    private fun exportStopwatchTimestamps(stopwatch: Stopwatch) {
```

Comment:
I think that one more function is needed: export all timestamps. It could be one of the options on the drop down list.